pkg/storaged/dialog.jsx

```diff
@@ -1007,9 +1007,13 @@ class SizeSliderElement extends React.Component {
             text_val = cockpit.format_number(val / unit);
         }
 
-        const change_unit = (_, u) => this.setState({
-            unit: Number(u),
-        });
+        const change_unit = (_, u) => {
+            if (val.unit)
+                onChange({ text: val.text, unit: Number(u) });
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19929-20240205-144049-d233f021-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)